### DISCUSSION TITLE
how to delete row including from another table linked to that said row (foreign)

### DISCUSSION THREAD

**Discussion by atiframlee** [supabase/supabase#1633] [Questions] [answered]
hi, newbies here. Im new to sql db

i have a table for ex, user table and address table with foreign key pointed to user table.
Let say if i want to delete the user and also address linked to that user in one go.

anyone care to show me how to implement that

**Comment by steve-chavez** [comment, marked as answer]
Hey @atiframlee,

For doing that, you need to define `on delete cascade` on the foreign key constraint.

```sql
create table addresses(
  id int primary key
, name text
, user_id int references users(id) on delete cascade
);
```

Then doing `delete from users where id = 1` will also delete the address, otherwise it would err.

If you already have your tables created, adding the `on delete cascade` would be like:

```sql
alter table addresses drop constraint <your_fk_name>;

alter table addresses add constraint <your_fk_name> foreign key (user_id)
references users(id) on delete cascade;
```

**Reply by atiframlee** [reply]
Thank you steve, ur guidance do help me out. bless you
